jscomp/others/js_null_undefined.mli

```diff
@@ -39,7 +39,30 @@ external null : 'a t = "null" [@@bs.val]
 (** The [undefined] value of type ['a Js.null_undefined] *)
 external undefined : 'a t = "undefined" [@@bs.val]
 
+(** [map f x] returns the result of running [f] on the value inside [x].
 
+If [x] is [null] or [undefined], returns it unchanged. If it contains a
+value, returns the result of calling [f] on that value, wrapped up as a
+nullable value again.
+
+@example {[
+map (fun [@bs] x -> x + 1) (return 1) = return 2
+map (fun [@bs] x -> x + 1) undefined = undefined
+]} *)
+val map : ('a -> 'b [@bs]) -> 'a t -> 'b t
+
```

Comment:
since we are going with `t` comes first rule.  `map` seems not needed

Reply:
Ah, I was going by the signatures in `Js.Option`, do you want to convert to the new style piecemeal? I'm scared that it will be a very confusing experience to use the different modules...

Reply:
yeah, it is true...